crates/storage/db-common/src/init.rs

```diff
@@ -160,10 +161,16 @@ pub fn insert_state<'a, 'b, DB: Database>(
 
     for (address, account) in alloc {
         let bytecode_hash = if let Some(code) = &account.code {
-            let bytecode = Bytecode::new_raw(code.clone());
-            let hash = bytecode.hash_slow();
-            contracts.insert(hash, bytecode);
-            Some(hash)
+            match Bytecode::new_raw_checked(code.clone()) {
+                Ok(bytecode) => {
+                    let hash = bytecode.hash_slow();
+                    contracts.insert(hash, bytecode);
+                    Some(hash)
+                }
+                Err(e) => {
+                    return Err(e);
```

Comment:
can we add an error trace for the account (address) here, this would make debugging easier

Reply:
In the original issue, @rakita mentioned 
> the error needs to be propagated as invalid json.

Do you know what this means?

Reply:
@danielcdz you can add an error trace with the [error!](https://docs.rs/tracing/latest/tracing/macro.error.html#examples) macro

Also you will need to transform the [EofDecodeError](https://github.com/emnul/reth/blob/4124144049a8c87c25d5ffb74718307df70aa512/crates/storage/errors/src/provider.rs#L19-L20) you get from the Bytecode::new_raw_checked function into a [ProviderError](https://github.com/emnul/reth/blob/4124144049a8c87c25d5ffb74718307df70aa512/crates/storage/errors/src/provider.rs#L19-L20) to get the code to compile. It looks like the `EofDecodeError` enum would need to be added as a member to the `ProviderError` enum, but this may lead to a breaking change and you would need to add the `revm-primitives` dependency to the `crates/storage/errors` crate so I would defer to a maintainer on the best approach.